### DISCUSSION TITLE
Meeting notes - 2021-11-19

### DISCUSSION THREAD

**Discussion by epijim** [insightsengineering/r-pkg-installation-validation#43] [Announcements]
# Meeting minutes

## Present

- James Black (Roche/Genentech)
- Doug Kelkhoff (Roche/Genentech)
- Tilo Blenk (GSK)

## Minutes

_There was no agenda in this initial discussion_

### Related to the _md template_

- Consensus that a detached approach of attaining user requirements, then documenting those being fulfilled, ran counter to how R packages are used. Instead focus can be on the documented functionality being sufficiently tested, and once that is confirmed CICD can be applied.
- General agreement that an initial assessment is needed to ensure the package has the documentation and especially unit testing present are meaningful
- Consensus that the current template, which documents every test against every function it tests is overly verbose. Should consider a summary table per function and use of an appendix, and to think through the stack depth we traverse looking for unit test scope.
- While we expect no differences on how an R package performs on Ubuntu head vs LTS, or RHEL vs CentOS, Roche stressed we would still require a build to work on our production validation systems 'just in time'
- Consensus that we could take a module approach where we identify what is the common ground we share, expecting that internally we may add additional testing, and against additional internal environments

### General discussion

- Covtracer was introduced, and could be leveraged beyond the _md template_. One example was to provide the traceability in an app designed to help reviewers.
- Stratification of risk is important. E.g. succesfully running unit tests should be sufficient for base R, and RStudio - but discussion on details of how stratification is applied is needed.
- Aspirational alignment on being able to go beyond shared 'components' of validation, and lighten the burden of validating CRAN together (but focus for now is starting small one what we can agree on for a simple report).

## Actions

- [x] @epijim to reach out to @dinakar29 and @andyofsmeg to see about transferring this repo over to pharmar
- [ ] @tlblnk (Tilo) to make issue about a simple summary table, and making it more accessible to reviewers

**Comment by cicdguy** [comment]
I can initiate the repository transfer from InsightsEngineering. An owner of `pharmaR` needs to simply approve the incoming transfer request.

**Reply by epijim** [reply]
Thanks Dinakar - sounds like there are some complications, or rather formalities needed by the R Consortium. So we can still aim to move their using your method (you will push it over and they just need to accept). 

But we'll hold off for now and I'll give Tilo, Andy and Ellis access in `inisightsengineering` for now.

**Reply by epijim** [reply]
Invites sent to Andy and Tilo (ellis already had access)

**Reply by epijim** [reply]
Note - proposal to R Validation Hub is here: https://github.com/insightsengineering/thevalidatoR/blob/contribution/.github/CONTRIBUTING.md#proposal-for-a-validation-documentation-sub-team-in-the-r-validation-hub

**Comment by epijim** [comment]
FYI -@tlblnk  is on holiday for a fortnight, so there will be a delay in kicking off tackling his ideas.
